aws/table_aws_backup_job.go

```diff
@@ -0,0 +1,242 @@
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsBackupJob(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_backup_job",
+		Description: "AWS Backup Job",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("job_id"),
+			Hydrate:    getAwsBackupJob,
+			Tags:       map[string]string{"service": "backup", "action": "DescribeBackupJob"},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAwsBackupJobs,
+			Tags:    map[string]string{"service": "backup", "action": "ListBackupJobs"},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(backupv1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "job_id",
+				Description: "The logical id of a backup job.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("BackupJobId"),
+			},
+			{
+				Name:        "recovery_point_arn",
+				Description: "An Amazon Resource Name (ARN) that uniquely identifies a recovery point.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("RecoveryPointArn"),
+			},
+			{
+				Name:        "backup_vault_arn",
+				Description: "An Amazon Resource Name (ARN) that uniquely identifies the target backup vault.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("BackupVaultArn"),
+			},
+			{
+				Name:        "resource_type",
+				Description: "The type of AWS resource to be backed up.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ResourceType"),
+			},
+			{
+				Name:        "resource_arn",
+				Description: "An Amazon Resource Name (ARN) that uniquely identifies the source resource in the recovery point.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ResourceArn"),
+			},
+			{
+				Name:        "status",
+				Description: "The current state of a backup job.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("State"),
+			},
+			{
+				Name:        "status_message",
+				Description: "A detailed message explaining the status of the job.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("StatusMessage"),
+			},
+			{
+				Name:        "backup_size",
+				Description: "The size in bytes of a backup.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("BackupSizeInBytes"),
+			},
+			{
+				Name:        "backup_vault_name",
+				Description: "The name of the target backup vault.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("BackupVaultName"),
+			},
+			{
+				Name:        "bytes_transferred",
+				Description: "The size in bytes transferred to a backup vault at the time that the job status was queried.",
```

Comment:
There are a few columns missing, for example (`BackupOptions`, `BackupType`). It would be great if you could add those as well.